lib/ReportConfig.php

```diff
@@ -55,14 +56,71 @@ public function getReportConfig() {
     }
   }
 
-  public function saveReportSummary($reportSummary) {
-      $config = $this->getReportConfig();
-      if (!$config) {
-        $this->module->setProjectSetting($this->reportConfigKey, array($reportSummary), $this->project_id);
+  /**
+   * Validate a ReportSummary.
+   * @param {Array} reportSummary - configuration for a report summary
+   */
+  public function validate($reportSummary) {
+    $errors = array();
+
+    $requiredFields = array('title', 'reportId', 'strategy');
+    foreach ($requiredFields as $requiredField) {
+      if (!array_key_exists($requiredField, $reportSummary)) {
+        $errors[] = "Missing field {$requiredField}";
+      }
+    }
+
+    if (array_key_exists('strategy', $reportSummary)) {
+      if (!in_array($reportSummary['strategy'], ReportStrategy::strategies())) {
+        $errors[] = 'Invalid strategy';
       } else {
-        $config[] = $reportSummary;
-        $this->module->setProjectSetting($this->reportConfigKey, $config, $this->project_id);
+        if ($reportSummary['strategy'] === ReportStrategy::ITEMIZED) {
+          if (array_key_exists('bucket-by', $reportSummary)
+              && strlen(trim($reportSummary['bucket-by'])) === 0) {
+            $errors[] = "bucket-by must have a value";
+          } else {
+            $errors[] = 'Missing bucket-by field when using strategy ' . ReportStrategy::ITEMIZED;
+          }
+        }
+      }
+    }
```

Comment:
It seems like if your UI code already does form validation that it would be redundant to have the backend be concerned with generating user-oriented messages. I think the backend should just validate and blow up if something unexpected comes through, and leave it to the vue.js code to prevent submission.

Reply:
I'm fine either way. It would make it easier to debug if we got a response back from the server that contained useful debug messages in the event there is drift between the UI validation and the backend validation.